coursera/api.py

```diff
@@ -455,6 +464,116 @@ def extract_links_from_exam(self, exam_id):
                 logging.exception('Could not download exam %s: %s', exam_id, exception)
             return None
 
+    def _get_notebook_folder(self, url, jupyterId, **kwargs):
+
+        supplement_links = {}
+
+        url = url.format(**kwargs)        
+        reply = get_page(
+            self._session,
+            url,
+            json=True
+        )
+
+        headers = self._auth_headers_with_json()
+
+        for content in reply['content']:
+            
+            if content['type'] == 'directory':
+                a = self._get_notebook_folder(OPENCOURSE_NOTEBOOK_TREE, jupyterId, jupId=jupyterId, path=content['path'], timestamp=int(time.time()))
+                supplement_links.update(a)
+            
+            elif content['type'] == 'file':
+                tmpUrl = OPENCOURSE_NOTEBOOK_DOWNLOAD.format(path=content['path'], jupId=jupyterId, timestamp=int(time.time()))
+                filename, extension = os.path.splitext(clean_url(tmpUrl))
+                
+                head, tail = os.path.split(content['path'])
```

Comment:
suggest path sanitizing here - remote paths can contain ":" that is ok on Linux but not allowed on Windows - something like this:
chars = '/?<>\:*|"'
table = maketrans(chars, '_' * len(chars))
path = path.translate(table)